src/main/resources/db/changesets/20200625_initial_schema.yaml

```diff
@@ -0,0 +1,143 @@
+    - createTable:
+        tableName: resource_type
+        columns:
+          - column:
+              name: id
+              type: uuid
+              constraints:
+                primaryKey: true
+                nullable: false
+          - column:
+              name: name
+              type: text
+              constraints:
+                nullable: false
+                unique: false
+    - createTable:
+        tableName: tracked_resource
+        columns:
+          - column:
+              name: id
+              type: uuid
+              constraints:
+                primaryKey: true
+                nullable: false
+          - column:
+              name: resource_type_id
+              type: uuid
+              constraints:
+                references: resource_type(id)
+                foreignKeyName: fk_type_id
+                nullable: false
+          - column:
+              name: resource_uid
+              type: json
+              constraints:
+                nullable: false
+                unique: false
+          - column:
+              name: creation
+              type: timestamp
+              constraints:
+                nullable: false
+                unique: false
+          - column:
+              name: expireation
+              type: timestamp
+              constraints:
+                nullable: false
+                unique: false
+          - column:
+              name: state
+              type: text
+              constraints:
+                nullable: false
+                unique: false
+          - column:
+              name: wait_until
+              type: timestamp
+              constraints:
+                nullable: false
+                unique: false
+    - createTable:
+        tableName: label
+        columns:
+          - column:
+              name: tracked_id
+              type: uuid
+              constraints:
+                references: tracked_resource(id)
+                foreignKeyName: fk_tracked_id
+                primaryKey: true
+                nullable: false
+          - column:
+              name: key
+              type: text
+              constraints:
+                nullable: false
+                unique: false
+          - column:
+              name: value
+              type: text
+              constraints:
+                nullable: false
+                unique: false
+    - createTable:
+        tableName: cleanup_log
+        columns:
+          - column:
+              name: tracked_id
+              type: uuid
+              constraints:
+                references: tracked_resource(id)
+                foreignKeyName: fk_tracked_id
+                primaryKey: true
+                nullable: false
+          - column:
+              name: logtime
+              type: timestamp
+              constraints:
+                nullable: false
+                unique: false
+          - column:
+              name: status
+              type: text
```

Comment:
Do we want to use an enum? Ditto for `tracked_resource.state`. I defer to others with practical db experience.
https://www.postgresql.org/docs/9.1/datatype-enum.html

Reply:
Historically, I have not used this.  We would either just use a `TEXT` field and maintain the enum in the code base, or extract the list of states into their own table and link them with a foreign key.  I think the issue ultimately boiled down to not wanting to maintain the enum in the code and the database and keep them in sync.  

That said, I don't really have a strong preference for whatever you choose to do here, I think it might be one of those situations where you can try it out and see how it works and you'll probably know pretty quickly how good or bad it looks.